libnmstate/nm/device.py

```diff
@@ -138,6 +147,20 @@ def _safe_activate_async(dev, connection_id):
     )
 
 
+def _get_connection(client, connection_id):
+    conn = None
+    if connection_id:
+        conn = client.get_connection_by_id(connection_id)
```

Comment:
This looks like we should have our own client class that just has a get_connection method that works like this.

Reply:
Eventiually it should be a  class that inherits from the NM client class. Moving it to a module is not an improvement IMHO. It does not need to be fixed now, though.